tools/advisor/advisor/db_log_parser.py

```diff
@@ -33,64 +36,87 @@ def is_new_log(log_line):
-
-    def get_time(self):
+        self.column_family = None
+        for col_fam in column_families:
+            search_for_str = '\[' + col_fam + '\]'
+            if re.search(search_for_str, self.message):
+                self.column_family = col_fam
+                break
+        if not self.column_family:
+            self.column_family = NO_FAM
+
+    def get_human_readable_time(self):
         return self.time
 
+    def get_column_family(self):
+        return self.column_family
+
     def get_context(self):
         return self.context
 
     def get_message(self):
         return self.message
 
     def append_message(self, remaining_log):
-        self.message = self.message + remaining_log
+        self.message = self.message + '\n' + remaining_log.strip()
+
+    def get_timestamp(self):
+        # assumes that the LOG timestamp is in GMT; this means that if this
+        # timestamp is to be converted to a human-readable value, one could
+        # either use the method get_human_readable_time() in this class or
+        # one can use 'datetime.utcfromtimestamp(<timestamp>).isoformat()';
+        # in either case the value returned would match the Log time in the LOG
+        # file
+        hr_time = self.time + 'GMT'
+        timestamp = timegm(time.strptime(hr_time, "%Y/%m/%d-%H:%M:%S.%f%Z"))
+        return timestamp
 
     def __repr__(self):
-        return 'time: ' + self.time + ', context: ' + self.context +\
-             ', message: ' + self.message
+        return (
+            'time: ' + self.time + '; context: ' + self.context +
+            '; col_fam: ' + self.column_family +
+            '; message: ' + self.message
+        )
 
 
 class DatabaseLogs(DataSource):
-    def __init__(self, logs_path_prefix):
+    def __init__(self, logs_path_prefix, column_families):
         super().__init__(DataSource.Type.LOG)
         self.logs_path_prefix = logs_path_prefix
+        self.column_families = column_families
 
-    def trigger_appropriate_conditions(self, conditions, log):
-        conditions_to_be_removed = []
+    def trigger_conditions_for_log(self, conditions, log):
         for cond in conditions:
             if re.search(cond.regex, log.get_message(), re.IGNORECASE):
-                cond.set_trigger(log)
-                conditions_to_be_removed.append(cond)
-        for remove_cond in conditions_to_be_removed:
-            conditions.remove(remove_cond)
-        return conditions
+                trigger = cond.get_trigger()
+                if not trigger:
+                    trigger = {}
+                if log.get_column_family() not in trigger:
+                    trigger[log.get_column_family()] = []
+                trigger[log.get_column_family()].append(log)
+                cond.set_trigger(trigger)
 
     def check_and_trigger_conditions(self, conditions):
         for file_name in glob.glob(self.logs_path_prefix + '*'):
+            if re.search('old', file_name, re.IGNORECASE):
```

Comment:
If this 'if-condition' is removed, then the code will check all the files in that directory.
I added this 'if-condition' because sometimes there are really old LOG files in the same directory which we don't want to consider.
I will add TODO for now, as you have suggested.